pkg/controller/controller_test.go

```diff
@@ -4470,6 +4840,92 @@ func generatePVCForProvisionFromPVC(srcNamespace, srcName, scName string, reques
 	return provisionRequest
 }
 
+// generatePVCForProvisionFromPVC returns a ProvisionOptions with the requested settings
+func generatePVCForProvisionFromXnsPVC(srcNamespace, srcName, scName, kind string, refpvcNamespace, apiGroup *string, requestedBytes int64, volumeMode string, setnamespace bool) controller.ProvisionOptions {
+	deletePolicy := v1.PersistentVolumeReclaimDelete
+
+	var provisionRequest controller.ProvisionOptions
+	if setnamespace {
+		provisionRequest = controller.ProvisionOptions{
```

Comment:
It looks like almost everything is the same and the only difference is the namespace field. I think you can move this out of this `if` block and only change the namespace inside.

Reply:
Because the test did not work with feature gate enabled when it was executed in parallel test.

Reply:
Fixed.